Webhooks features 2.0

italojmf opened the pull request:
Web Hooks Features: 
* Adapted to work with bbb-2.0
* Map internal bbb messages to external messages to easily build an API to receive events;
* Permanent hook:
 Can't be deleted via API;
 Won't expire even if callback fails for a long time;
 URL defined in configs;
 Global;
 Receive either raw¹ or mapped² messages (configurable);
* Send up to X (configurable) messages on a single request (when they are simultaneously at the same queue);
* Limit queue to a defined size;
* Add new parameter to hooks/create: "getRaw=true/false" allowing hooks to receive raw messages instead of mapped ones;
* Each hook has it's own event queue and emit events asynchronously;
* Map userID;
* Update packages;
* Convert to JS ES6;
* "auth2_0" (configurable : true = use this method, false = use checksum method): Send data as a JSON with a bearer authorization (which is the shared secret), to the defined callbackURL, without the checksum.
 
The "config.js" file is initially set in a way it'll not use the features until properly configured, to allow compatibility, which means created hooks will get:
* Raw messages;
* Only 1 message per POST;
* Use checksum auth.
  
¹ - a message in the exact format sent by BBB.
² - few preset messages that are mapped to a different format, available at: http://docs.mconf.org/apis/webhooks-api/ .

daronco commented on bbb-webhooks/callback_emitter.js:
```diff
@@ -0,0 +1,109 @@
+          } else {
+            this.nextInterval = !this.permanent ? 0 : 8; // Reset interval to permanent hooks
+            // If a hook has backup URLs for the POSTS, use them after a few failed attempts
+            if ((this.backupURL != null) && this.permanent) { this.backupURL.push(this.backupURL[0]); this.backupURL.shift(); this.callbackURL = this.backupURL[0]; }
```
This block is a bit confusing. It could be rewritten as:

```js
if (this.permanent) {
  if (this.backupURL != null) {
    this.backupURL.push(this.backupURL[0]);
    this.backupURL.shift();
    this.callbackURL = this.backupURL[0];
  }
} else {
  return this.emit("stopped");
}
```

daronco commented on bbb-webhooks/callback_emitter.js:
```diff
@@ -0,0 +1,109 @@
+
+          // no intervals anymore, time to give up
+          } else {
+            this.nextInterval = !this.permanent ? 0 : 8; // Reset interval to permanent hooks
```
This `8` could be in the config file.

daronco commented on bbb-webhooks/callback_emitter.js:
```diff
@@ -0,0 +1,109 @@
+    };
+
+    request(requestOptions, function(error, response, body) {
+      if ((error != null) || !(((response != null ? response.statusCode : undefined) >= 200) && ((response != null ? response.statusCode : undefined) < 300))) {
```
You could move the part that checks the `response` value to another method to make this `if` simpler.

daronco commented on bbb-webhooks/config.js:
```diff
@@ -0,0 +1,96 @@
+const config = require("./config_local.js");
+
+// BigBlueButton configs
+if (!config.bbb) { config.bbb = {}; }
```
These comparisons can have unexpected results. Try this for example:

```
var config = {};
config.port = 0;
(!config.port) ? 'not defined' : 'defined';
```

It will print 'not defined', even though the property `port` was defined.
Use instead [`config.hasOwnProperty('port')`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty), that will return true no matter what value the property was set to, or `config.bbb == null` that will be true if it's `null` or `undefined`.

daronco commented on bbb-webhooks/config.js:
```diff
@@ -0,0 +1,96 @@
+    rapChannel: 'bigbluebutton:from-rap'
+  }
+ }
+// IP where aggr will be hosted
```
Don't call this `aggr`, the webhooks shouldn't care that the events will be received by this component. This is an array or permanent hooks, so could be called something like `config.hooks.permanent`.

daronco commented on bbb-webhooks/config_local.js.example:
```diff
@@ -0,0 +1,27 @@
+config.hooks.channels = {
+  mainChannel: 'from-akka-apps-redis-channel',
+  rapChannel: 'bigbluebutton:from-rap',
+  chatChannel: 'from-akka-apps-chat-redis-channel'
```
The `chatChannel` is not on `config.js`, is that on purpose or should it be a default channel to?

daronco commented on bbb-webhooks/extra/events.js:
```diff
@@ -15,32 +15,25 @@ subscriber.on("psubscribe", function(channel, count) {
-
-      var message_meeting_id = message.payload.meeting_id;
-      var message_name = message.header.name;
+    if (message && message.envelope) {
```
Be careful with these comparisons, it's better to use `hasOwnProperty` or `!= null`.

daronco commented on bbb-webhooks/hook.js:
```diff
@@ -0,0 +1,333 @@
+
+  save(callback) {
+   this.redisClient.hmset(config.redis.keys.hook(this.id), this.toRedis(), (error, reply) => {
+      if (error != null) { Logger.error("[Hook] error saving hook to redis:", error, reply); }
```
Take a look at the indentation here and in other places in this file. Some lines are off by 1 space.

daronco commented on bbb-webhooks/hook.js:
```diff
@@ -0,0 +1,333 @@
+  // if there is no emitter running already and if there is a message in the queue.
+  _processQueue() {
+    // Will try to send up to 10 messages together if they're enqueued
+    const lengthIn = this.queue.length > 10 ? 10 : this.queue.length;
```
This `10` could be in `config.js`.

daronco commented on bbb-webhooks/hook.js:
```diff
@@ -0,0 +1,333 @@
+    if ((message == null) || (this.emitter != null) || (lengthIn <= 0)) { return; }
+    // Add params so emitter will 'know' when a hook is permanent and have backupURLs
+    this.emitter = new CallbackEmitter(this.callbackURL, message, this.backupURL);
+    this.emitter.start(this.permanent);
```
I think it makes more sense to pass the list of permanent hooks when creating the `CallbackEmitter` instead of here in the `start`.

daronco commented on bbb-webhooks/hook.js:
```diff
@@ -0,0 +1,333 @@
+          if (error != null) { Logger.error("[Hook] error pushing event to redis queue:", JSON.stringify(message), error); }
+        });
+        this.queue.push(JSON.stringify(message));
+        if (!hold) { this._processQueue(); }
```
Is this `hold` parameter used anywhere else other than in the tests? Could only find it being set in the tests.
If it's not used in the code it should exist. There's probably another way to do it in the tests that doesn't require changing the code.

daronco commented on bbb-webhooks/hook.js:
```diff
@@ -0,0 +1,333 @@
+      this._processQueue(); // go to the next message
+    });
+
+    // gave up trying to perform the callback, remove the hook forever if the hook's not permanent (emmiter will validate that)
```
Is anyone really validating it? I look at `destroy` and it doesn't check if the hook is permanent or not.

italojmf replied:
```destroy``` calls ```Hook.removeSubscription``` which validade if the hook is or not permanent

daronco commented on bbb-webhooks/hook.js:
```diff
@@ -0,0 +1,333 @@
+
+  static removeSubscription(hookID, callback) {
+    let hook = Hook.getSync(hookID);
+    if ( hook != null && ((hook.permanent === "false") || (hook.permanent === false))) {
```
Why compare with "false"? Would be better if `permanent` was always a boolean.

daronco commented on bbb-webhooks/hook.js:
```diff
@@ -0,0 +1,333 @@
+    Hook.resync(callback);
+  }
+
+  static flushall() {
```
`flushall` and `flushredis` are only used on testes, so they should be there, in the tests, not here.

daronco commented on bbb-webhooks/config_local.js.example:
```diff
@@ -0,0 +1,27 @@
+config.server.port = 3005;
+
+// Callbacks will be triggered for all the events in this list and only for these events.
+config.hooks = {};
```
Could leave this `config.hooks` and all other things inside of them commented.

daronco reviewed (changes requested):
Other than what was commented on the files, please make these other changes:

* Update `.nvmrc` to the version of node being used now (as it was updated on `package.json` already).
* Add information about the tests in the README file
* The tests should not depend on having a BigBlueButton server running. They can depend on redis, though, so you can insert events on redis to simulate what BigBlueButton would do.
* Merge mconf:bbb-2x-mconf again to resolve the conflicts.
* If you don't set any permanent hook, it will create one `undefined`, but it should create any hook.

antobinary committed:
Merge pull request #4734 from OZhurbenko/presenter-functionality-2x

Whiteboard touch events support, bug fixes, improvements

prlanzarin committed:
Merge remote-tracking branch 'gutobenn/bbb-webrtc-sfu' into bbb-webrtc-sfu

KDSBrowne committed:
make userlist accessibility work via props/state

lfzawacki committed:
Adjustments to video reconnect code

antobinary committed:
Merge pull request #4687 from Gcampes/audio-refactor-bugfixes

Audio bugfixes and improvements

capilkey committed:
add a microphone warning for chrome on http

lfzawacki committed:
Merge branch 'bbb-webrtc-sfu' of github.com:prlanzarin/bigbluebutton-1 into bbb-webrtc-sfu

Conflicts:
	bigbluebutton-html5/imports/ui/components/video-dock/component.jsx

capilkey committed:
block webcam popup opening when locked

antobinary committed:
Merge pull request #4662 from Tainan404/fix-design

Fix design issues

KDSBrowne committed:
clean up code

OZhurbenko committed:
Disabled spellcheck for the text shape and added an android check

.. so that tap-to-focus text shape workaround for FF 57 and 58 for Android didn't interfere with the desktop versions

KDSBrowne committed:
add prop validation

KDSBrowne committed:
Merge remote-tracking branch 'upstream/v2.0.x-release' into z14-fix-userlist-keyboard-accessibility

KDSBrowne committed:
remove focus when ESC key pressed

OZhurbenko committed:
Fixed Android detection for the text shape

capilkey committed:
added poll record events, removed unused HidePoll message

Tainan404 committed:
removed click error and userlist item that remained selected

Commit:
Remove unused package and fix whiteboard toolbar submenu tooltip

ritzalam committed:
Merge pull request #4776 from capilkey/add-poll-record-events

Add poll record events, remove unused HidePoll message

capilkey committed:
(2.0 only) block zoom slider change processing when not presenter

capilkey committed:
check for failed chat message parse

antobinary committed:
Merge pull request #4712 from Tainan404/localize-public-clear

 Localize message from when clear the chat.

antobinary committed:
handle join and leave of call-in user

antobinary committed:
Protect dial-in users from web user actions

disallow { promote / demote / make presenter / clear status } being triggered on a dial in user

antobinary committed:
linting and clean-up

ffdixon committed:
Merge pull request #4780 from capilkey/2.0-block-zoom-slider-change

(2.0 only) block zoom slider change processing when not presenter

Commit:
Remove log and add delay between hover/focus and showing the tooltip

prlanzarin committed:
Changed webcam localization strings

ritzalam committed:
Merge pull request #4754 from antobinary/join-via-html5-over-flash

Provide option to join via html5 over flash client

ritzalam committed:
Merge pull request #4760 from capilkey/chrome-microphone-access-warning

Add a warning about microphone access for chrome on HTTP

ritzalam committed:
Merge pull request #4761 from capilkey/fix-webcam-shortcut-lock

Block webcam popup opening when locked

prlanzarin committed:
Removed unused Screenshare import

ritzalam committed:
Merge pull request #4781 from capilkey/check-chat-message-parse-fail

Check for failed chat message parse

prlanzarin committed:
Removed unused methods from actions bar component

prlanzarin committed:
Removed unused states from video-dock component and refactored event listeners

prlanzarin committed:
Moved video-dock and video-menu collection communication to services

antobinary committed:
add ejectUserFromVoice for moderator

prlanzarin committed:
Refactored webcam share/unshare handlers

prlanzarin committed:
Removed router for video-menu

Also removed unused class in video-dock css

antobinary committed:
Merge pull request #4769 from OZhurbenko/presenter-functionality-2x

Fixed Text Shape for the desktop FF 57 and 58, disabled spellCheck

antobinary committed:
Merge pull request #4751 from Tainan404/fix-userlist-bug

 Fix false click in userlist.

Tainan404 committed:
Merge remote-tracking branch 'upstream/v2.0.x-release' into toast-breakout-ends

Commit:
Hide labels from actions, status, mute/unmute and join/leave audio buttons

KDSBrowne committed:
remove unused user-list-header

jfederico committed:
bbb-lti: Rework on the way XML is parsed in order to fix issue with getRecording

ffdixon committed:
Merge pull request #4794 from jfederico/v2.0.x-release_bbb-lti-fix-getrecordings

bbb-lti: Rework on the way XML is parsed in order to fix issue with getRecordings

jfederico committed:
bbb-lti: Fixed dataTable and made the eye icon toggle

Tainan404 committed:
fix merge conflict

Commit:
Change actions button icon

Commit:
Resolve linter problems

jfederico committed:
bbb-lti: Added image preview to recordings

ffdixon committed:
Merge pull request #4796 from jfederico/v2.0.x-release_bbb-lti-fix-getrecordings

bbb-lti: Fixes datatable and adds image preview

Tainan404 committed:
add screen to when the meeting is closed

OZhurbenko committed:
Fixed sorting in the user list

OZhurbenko committed:
Fixed bottom padding for the last chat message in Firefox and Edge

antobinary committed:
poll voted - drop requesterId and rename event

MaximKhlobystov committed:
Merge pull request #4802 from OZhurbenko/fix-message-list-bottom-padding

Fixed last chat message view in Firefox

OZhurbenko committed:
Fixes a default locale selection for Edge, Windows 10

OZhurbenko committed:
Merge branch 'v2.0.x-release' of https://github.com/bigbluebutton/bigbluebutton into fix-userlist-sorting

OZhurbenko committed:
Corrected the detection of the clear_status

antobinary committed:
Merge pull request #4800 from OZhurbenko/fix-userlist-sorting

Fixed sorting in the user list

antobinary committed:
Merge pull request #4804 from OZhurbenko/fix-edge-default-locale

Fixed a problem with the default locale selection in Edge

oswaldoacauan committed:
Remove duplicated log

antobinary committed:
Move <audio> to main.html

Currently it is deep within Base component. So when Base is re-rendered with the updated locale, we lose <audio>. Placing it in main.html, outside Base resolved the problem

oswaldoacauan committed:
Throw meteor error when Acl fails for subscriptions

oswaldoacauan committed:
Simplify auth code and remove clearCredentials on errors. Fix #4812

oswaldoacauan committed:
Add redirect to /logout if not loggedin or on subscription errors

antobinary committed:
Merge pull request #4683 from KDSBrowne/z14-fix-userlist-keyboard-accessibility

Fix user-list keyboard accessibility

oswaldoacauan committed:
Fix Meteor.error subscription params

Tainan404 committed:
fix error in logoutURL

Tainan404 committed:
add more width for modal

antobinary committed:
audio related linting

antobinary committed:
Merge pull request #4753 from jfsiebel/html5-tooltip

Add tooltip for buttons without label

antobinary committed:
set defaultProps for glow to be false

antobinary committed:
add permissionLayout location for Safari

Commit:
Merge remote-tracking branch 'upstream/v2.0.x-release' into hide-audio-labels

antobinary committed:
Merge pull request #4803 from antobinary/adjust-polls-4778

Poll updated event - drop requesterId/presenterId and rename event

antobinary committed:
Update index_html5_vs_flash.html

antobinary committed:
Merge pull request #4822 from bigbluebutton/antobinary-update-index.html

Update index_html5_vs_flash.html

KDSBrowne committed:
fix console warnings

KDSBrowne committed:
remove console logs

lfzawacki committed:
Capture several errors and crashes in client and bbb-webrtc-sfu

Tainan404 committed:
add meetingEnded in Base component aand add intl translation

lfzawacki committed:
Merge branch 'bbb-webrtc-sfu' of github.com:prlanzarin/bigbluebutton-1 into bbb-webrtc-sfu

Removed several variables that were in this.state to this in the html5 video-dock component

Conflicts:
	bigbluebutton-html5/imports/ui/components/video-dock/component.jsx

KDSBrowne committed:
convert to component that entends Purefunction

KDSBrowne committed:
fix Failed prop type warning in LeaveConfirmation

oswaldoacauan committed:
Merge remote-tracking branch 'upstream/v2.0.x-release' into issue-4812

Commit:
Alter status tooltip to match selected status

KDSBrowne committed:
Merge remote-tracking branch 'origin/x.0.2-fix-invalid-prop' into x.0.3-fix-console-warnings

KDSBrowne committed:
clean up code

KDSBrowne committed:
hide fullscreen option for iOS devices

KDSBrowne committed:
remove console log

antobinary committed:
Merge pull request #4788 from antobinary/audio-work-3

Handle dial-in users in HTML5

lfzawacki committed:
Don't send the 'media not flowing' event until we figure things out

KDSBrowne committed:
fix failed prop type: scrollPosition in Chat component

KDSBrowne committed:
fix remaining console warnings

oswaldoacauan committed:
Fix case when there is no current selected

OZhurbenko committed:
Fixed round buttons' icon centering

Fixes #3576

OZhurbenko committed:
Changed a default tool to Pencil

Fixes #4807

OZhurbenko committed:
Disabling the Thickness and Colour menu items when the 'hand' tool is selected

Fixes #4808

antobinary committed:
Merge pull request #4830 from OZhurbenko/fix-issue-3576

Fixed round buttons' icon centering

antobinary committed:
Merge pull request #4816 from jfsiebel/hide-audio-labels

Hide audio labels

antobinary committed:
Merge pull request #4733 from oswaldoacauan/fix-4700

Improvements on Presentation Management

KDSBrowne committed:
add button label

KDSBrowne committed:
move _.uniqueId() out of render function

KDSBrowne committed:
remove new line between imports

KDSBrowne committed:
add comment to actions-dropdown component

KDSBrowne committed:
fix client not loading on iOS devices

Tainan404 committed:
fix intl id

Tainan404 committed:
change intl id

oswaldoacauan committed:
Fix 401 on page refresh

MaximKhlobystov committed:
Convert the stateless Chat component into a class component.

MaximKhlobystov committed:
Made sure that when we open a new chat, it is not on the list of closed ones anymore.

MaximKhlobystov committed:
Small cleanup - removed unnecesssary import statement.

MaximKhlobystov committed:
Linting.

MaximKhlobystov committed:
Merge branch 'v2.0.x-release' of https://github.com/bigbluebutton/bigbluebutton into reopen-private-chat-bug

MaximKhlobystov committed:
Converted Chat component back to its stateless version.

MaximKhlobystov committed:
Made Chat container a class component

MaximKhlobystov committed:
Applied the private chat fix to Chat container.

MaximKhlobystov committed:
Small comment.

ritzalam committed:
- fix issue https://github.com/bigbluebutton/bigbluebutton/issues/4660

ritzalam committed:
- hardcode logoutTimer to zero as we don't use this feature in the release
 - substitute keyworks in moderatorOnlyMessage

Tainan404 committed:
fix settings for mobile screen

oswaldoacauan committed:
Fix bug when users re-validate after refresh

KDSBrowne committed:
add comment

lfzawacki committed:
Refactor html5 video tag creation to work with React style

lfzawacki committed:
Move WS open/close events to component mount/unmount

prlanzarin committed:
Merge pull request #89 from lfzawacki/html5-development-reconnect-issues

HTML5 Videos Reconnect issues

lfzawacki committed:
Tweaks on the video media flow

ritzalam committed:
Merge pull request #4843 from ritzalam/fix-issue-4660

 - fix issue https://github.com/bigbluebutton/bigbluebutton/issues/4660

ritzalam committed:
Merge pull request #4844 from ritzalam/fix-logouttimer-and-mod-only-msg

 - hardcode logoutTimer to zero as we don't use this feature in the r…

GhaziTriki committed:
Fix put meeting and at muted status.

lfzawacki committed:
Small fixes on the server and iceCandidate sending

lfzawacki committed:
Better video stop treatment in HTML5

prlanzarin committed:
Changing the way the unshare webcam API is called to support user roles

prlanzarin committed:
Fixed video session management on bbb-webrtc-sfu

This should solve the most severe cases of ghost webcams

ritzalam committed:
- allow passing of custom param (logo) to override default logo

lfzawacki committed:
Merge remote-tracking branch 'prlanzarin1/ghost-webcams' into html5-development-reconnect-issues

Conflicts:
	bigbluebutton-html5/imports/ui/components/video-dock/component.jsx
	labs/bbb-webrtc-sfu/lib/video/VideoManager.js
	labs/bbb-webrtc-sfu/lib/video/video.js

prlanzarin committed:
Made the users own webcam local instead of server-side

lfzawacki committed:
Use more readable uids for bbb-webrtc-sfu

lfzawacki committed:
Merge remote-tracking branch 'prlanzarin1/ghost-webcams' into html5-video-media-flow

Conflicts:
	bigbluebutton-html5/imports/ui/components/video-dock/component.jsx

ritzalam committed:
- users come in as always unmuted, so no need to check muted flag

ritzalam committed:
Merge branch 'riadvice-fix-mute-meeting' into v2.0.x-release

ritzalam committed:
Merge pull request #4852 from ritzalam/all-custom-log

 - allow passing of custom param (logo) to override default logo

ritzalam committed:
- allow to pass custom copyright text using create API

ritzalam committed:
Merge pull request #4857 from ritzalam/custom-copyright

 - allow to pass custom copyright text using create API

prlanzarin committed:
Merge remote-tracking branch 'upstream/v2.0.x-release' into video-reconnection

prlanzarin committed:
Added checking for HTML5 webcams

This should prevent blank webcams on the flash client

prlanzarin committed:
Tooltips for video buttons and actions-bar container service calls fixes

ritzalam committed:
- implement muteOnstart by passing on create API or using bigbluebutton.properties in bbb-web instead of setting it in config.xml

ritzalam committed:
Merge pull request #4861 from ritzalam/set-mute-on-start

 - implement muteOnstart by passing on create API or using bigbluebut…

ritzalam committed:
- change all kick works to remove

ritzalam committed:
Merge branch 'v2.0.x-release' of github.com:bigbluebutton/bigbluebutton into change-ejected-msg

ritzalam committed:
Merge pull request #4862 from ritzalam/change-ejected-msg

Change ejected msg

capilkey committed:
add a minute warning when breakout rooms are going to close

ritzalam committed:
Merge branch 'breakout-room-closing-warning' of https://github.com/capilkey/bigbluebutton into capilkey-breakout-room-closing-warning

ritzalam committed:
- remove trace log

ritzalam committed:
Merge branch 'capilkey-breakout-room-closing-warning' into v2.0.x-release

prlanzarin committed:
Removed wrong video peer stop procedure and reset the kurento.yaml config file

prlanzarin committed:
Merge pull request #92 from prlanzarin/video-reconnection

Ghost webcams and merging with upstream

antobinary committed:
Merge branch 'bbb-webrtc-sfu' of github.com:prlanzarin/bigbluebutton-1 into test-4661

antobinary committed:
Merge pull request #4661 from prlanzarin/bbb-webrtc-sfu

Add support for WebRTC screensharing into Flash/HTML5 and WebRTC video into HTML5

lfzawacki committed:
Put rid dependency in production part of package.json

lfzawacki committed:
Put rid dependency in production part of package.json

lfzawacki committed:
Merge pull request #93 from lfzawacki/html5-development

Put rid dependency in production part of package.json

prlanzarin committed:
Revert "Added checking for HTML5 webcams"

This reverts commit 51b62ed30d745ea45e96ae31958d7aa71d00c8e1.

prlanzarin committed:
Filter out html5 cameras from flash client

antobinary committed:
add muteOnStart param in Meeting structure

ritzalam committed:
Merge pull request #4866 from antobinary/test-4661

Adjust Meeting structure to contain new property muteOnStart

OZhurbenko committed:
Merge pull request #4867 from lfzawacki/bbb-webrtc-sfu

Fix problem when building HTML5 screenshare + video PR

GhaziTriki committed:
Fix toolbar components alignment and localisation update.

GhaziTriki committed:
Fix setters in Meeting.java

GhaziTriki committed:
Disable private chat in breakout rooms.

GhaziTriki committed:
Add a new configuration option `privateChateEnabled` for breakout rooms.

ffdixon committed:
Merge pull request #4871 from riadvice/breakout-room-fix-v2.0

Disable private chat for breakout rooms

antobinary committed:
Merge pull request #4814 from oswaldoacauan/issue-4812

Simplify authentication code/flow and Add subscriptions error handling

antobinary committed:
Merge pull request #4818 from antobinary/fix-ios-audio

Add permissionLayout location for Safari

antobinary committed:
Merge pull request #4831 from OZhurbenko/presenter-functionality-2x

Improved Whiteboard toolbar's user experience

KDSBrowne committed:
Merge remote-tracking branch 'upstream/v2.0.x-release' into x.0.3-fix-console-warnings

KDSBrowne committed:
remove import not being used

prlanzarin committed:
Finished filtering out HTML5 and Flash webcams using stream name patterns

OZhurbenko committed:
Changed the message displayed to the kicked user

Fixes #4863

OZhurbenko committed:
Updated the html5 client's font to use the latest icons

Fixes #4715

antobinary committed:
Merge pull request #4624 from Tainan404/toast-breakout-ends

[HTML5] show audio modal and toast notification when breakout room ends

antobinary committed:
Merge pull request #4740 from Tainan404/fix-dropdown-bug

Fix user list dropdown bug when making user presenter.

antobinary committed:
Merge pull request #4884 from OZhurbenko/fix-issue-4715

Updated the html5 client's font to use the latest icons

antobinary committed:
Merge pull request #4883 from OZhurbenko/fix-issue-4863

Updated the message displayed to the kicked user

prlanzarin committed:
Refactored the flash webcam pattern checking

prlanzarin committed:
Removed *sw* entry from .gitignore

gutobenn committed:
Error notifications for video and screenshare

lfzawacki committed:
Detect online/offline browser and open/close WS accordingly

gutobenn committed:
Localize video and screnshare error messages

GhaziTriki committed:
Change webcamsOnlyForModerator property and handle saving it in server and client.

GhaziTriki committed:
Add missing class SetWebcamsOnlyForModeratorEvent.

GhaziTriki committed:
Request webcamsOnlyForModerator value at user enter success.

OZhurbenko committed:
Switched from the broken 4commerce:env-settings package to native Meteor.settings

OZhurbenko committed:
Linting

OZhurbenko committed:
Switched the components from deprecated  method to

OZhurbenko committed:
Updated Meteor to 1.6.0.1

GhaziTriki committed:
Sort private chat users alphabetically.

italomfranco committed:
Change preset config for compatibility

antobinary committed:
Merge pull request #4914 from OZhurbenko/switch-to-meteor-settings

Moved settings away from a broken 4commerce:env-settings package

ritzalam committed:
Merge pull request #4896 from prlanzarin/flash-webcam-fix

Ghost flash webcams fix

ritzalam committed:
Merge pull request #4916 from riadvice/alphabetise-private-chat-users

Sort private chat users alphabetically.

ritzalam committed:
Merge pull request #4868 from riadvice/2.0-ui-fixes

Fix toolbar components alignment and localisation update

GhaziTriki committed:
Merge remote-tracking branch 'bigbluebutton/v2.0.x-release' into webcams-only-for-moderator-on-off

GhaziTriki committed:
Revert webcamsOnlyForModerator feature in VideoEventMapDelegate.

antobinary committed:
Merge pull request #4915 from OZhurbenko/update-meteor-16

Updated Meteor to 1.6.0.1

antobinary committed:
Merge pull request #4841 from MaximKhlobystov/reopen-private-chat-bug

[HTML5 Client] Fix for reopening private chat after using X button

antobinary committed:
Merge pull request #4832 from KDSBrowne/x.0.3-fix-console-warnings

Fix Invalid prop 'children[2]' supplied to `DropdownList`

antobinary committed:
Merge pull request #4833 from KDSBrowne/x.0.4-ios-hide-fullscreen

Hide fullscreen option for iOS devices

ritzalam committed:
- fix issue where muteOnStart doesn't work when set in bigbluebutton.properties

ritzalam committed:
Merge pull request #4922 from ritzalam/mute-on-start

 - fix issue where muteOnStart doesn't work when set in bigbluebutton…

GhaziTriki committed:
Automatically subscribe/unsubcribe to video streams when webcamsOnlyForModerator value changes.

GhaziTriki committed:
Remove webcamsOnlyForModerator from enter API response.

GhaziTriki committed:
Set webcamsOnlyForModerator in MeetingActor when the meeting is creator.

antobinary committed:
Merge branch 'v2.0.x-release' into fix-audio-locale-change

ritzalam committed:
Merge pull request #4925 from riadvice/webcams-only-for-moderator-on-off

Switch webcamsOnlyForModerator on and off

antobinary committed:
remove AudioComponent as we do not render anything

lfzawacki committed:
Fix problems on video tags if camera is busy

Tainan404 committed:
Merge remote-tracking branch 'upstream/v2.0.x-release' into fix-settings-mobile

Tainan404 committed:
remove multiplication for 1

lfzawacki committed:
Wait one tick before resizing HTML5 videos

gutobenn committed:
Prevent multiple clicks on share/unshare video button

Tainan404 committed:
Merge remote-tracking branch 'upstream/v2.0.x-release' into fix-moderator-ends

GhaziTriki committed:
Fix alpha value for ScreenshareViewWindow change display button.

ffdixon committed:
Merge pull request #4943 from riadvice/2.0-ui-fixes

Fix alpha value for ScreenshareViewWindow change display button

lfzawacki committed:
Better clean up on the bbb-wbertc serverside to fix html5 video problems

lfzawacki committed:
Prevent errors on html5 video when socket disconnects

antobinary committed:
Merge pull request #4815 from antobinary/fix-audio-locale-change

Fix audio loss on locale change

antobinary committed:
Merge pull request #4801 from Tainan404/fix-moderator-ends

Add screen to when the meeting is closed.

GhaziTriki committed:
Show moderator message header in bold.

GhaziTriki committed:
Fix ScreenshareViewWindow scroll-bars in original video size mode.

ffdixon committed:
Merge pull request #4946 from riadvice/2.0-ui-fixes

Fix ScreenshareViewWindow scroll-bars in original video size mode

GhaziTriki committed:
Delay showing configramtion buttons in WebRTCEchoTest under firefox browser.

ffdixon committed:
Merge pull request #4953 from riadvice/fix-firefox-webrtc

Delay showing confirmation buttons in WebRTCEchoTest under firefox

ritzalam committed:
Merge pull request #4945 from riadvice/moderator-message-ui

Show moderator message header in bold

prlanzarin committed:
Fixed incorrect fetching of userId in the video-dock component

prlanzarin committed:
Removed unused screenshare container from videodock and fixed stop message userId

lfzawacki committed:
Use service joinedVideo in the correct stage of video sharing flow

prlanzarin committed:
Merge remote-tracking branch 'mconftec/fixes-lucas' into video-upstream-errors

prlanzarin committed:
Merge remote-tracking branch 'mconftec/disable-share-webcam-button' into video-upstream-errors

prlanzarin committed:
Merge branch 'v2.0.x-release' of https://github.com/bigbluebutton/bigbluebutton into video-upstream-errors

GhaziTriki committed:
Disable webcam button in media column for all users when webcamsOnlyForModerator is on.

ffdixon committed:
Merge pull request #4957 from riadvice/webcams-only-for-moderator-on-off

Disable webcam button in media column for all users when webcamsOnlyForModerator is on

antobinary committed:
Merge pull request #4846 from Tainan404/fix-settings-mobile

Change modal settings styles.

antobinary committed:
Merge pull request #4955 from prlanzarin/video-upstream-errors

WebRTC media updates

ffdixon committed:
Update settings-production.json when assigning hostname

ffdixon committed:
Merge pull request #4965 from ffdixon/update-bbb-conf

Update settings-production.json when assigning hostname

italomfranco committed:
Merge remote-tracking branch 'upstream/v2.0.x-release' into webhooks-features-2.0

ghost commented:
Will be sent to https://github.com/mconftec/bigbluebutton/tree/html5-development.
